### PR TITLE
[MIRROR] Add clickable action buttons from /TG/, rework a lot of backend and other improvements.

### PR DESCRIPTION
**Pull request by VirgoBot**
Original PR: https://github.com/PolarisSS13/Polaris/pull/7315
--------------------
- Gygax overload, Durand defence mode, phazon phase, marauder smoke, zoom and thrusters have been made into modular things. Just put the 'smoke_possible' var to get smoke on your custom thing for example. This should work for custom map spawns, but that hasn't been properly tested yet.
- Move most abilities that aren't for default mechs to 'Mecha_actions.dm' for a more reasonable and not so huge file. 
  Marauder flight to be moved shortly.
- Lot of verbs have been broken into procs such as toggle_lights() now being a verb refering to lights() for easier work later on.
- Delete some movement code duplication with the gygax, gorilla, durand and phazon.
- Changed the smoke system to be more up to date, inspiration taken from the cooking machines.
- Added a noise for the internal air tank usage.
- Improved on the ripley speedbooster module code.
- Even added COMMENTS

image

### CODE REVIEW

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -100,8 +102,62 @@
...
+	var/datum/action/innate/mecha/mech_smoke/smoke_action = new
+	var/datum/action/innate/mecha/mech_zoom/zoom_action = new
+	var/datum/action/innate/mecha/mech_toggle_thrusters/thrusters_action = new
+	var/datum/action/innate/mecha/mech_cycle_equip/cycle_action = new
+	var/datum/action/innate/mecha/mech_switch_damtype/switch_damtype_action = new
+	var/datum/action/innate/mecha/mech_toggle_phasing/phasing_action = new
+
+
+
+>>>>>>> ce9ceb4... Add clickable action buttons from /TG/, rework a lot of backend and other improvements. (#7315)
```
Comment (lbnesquik) [inline code/game/mecha/mecha.dm:160]: Here's the issue bby

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -1345,8 +1475,31 @@
 		src.verbs += /obj/mecha/verb/eject
 		src.log_append_to_last("[H] moved in as pilot.")
 		src.icon_state = src.reset_icon()
+<<<<<<< HEAD
```
Comment (lbnesquik) [inline code/game/mecha/mecha.dm:1478]: Another thing to remove

File: code/game/mecha/mecha.dm [hunk 3]
```diff
@@ -1345,8 +1475,31 @@
 		src.verbs += /obj/mecha/verb/eject
 		src.log_append_to_last("[H] moved in as pilot.")
 		src.icon_state = src.reset_icon()
+<<<<<<< HEAD
 		if(occupant.hud_used)
 			minihud = new (occupant.hud_used, src)
+=======
```
Comment (lbnesquik) [inline code/game/mecha/mecha.dm:1481]: Remov

File: code/game/mecha/mecha.dm [hunk 4]
```diff
@@ -1345,8 +1475,31 @@
...
+			verbs -= /obj/mecha/verb/toggle_overload
+		if(!zoom_possible)
+			verbs -= /obj/mecha/verb/toggle_zoom
+		if(!phasing_possible)
+			verbs -= /obj/mecha/verb/toggle_phasing
+		if(!switch_dmg_type_possible)
+			verbs -= /obj/mecha/verb/switch_damtype
+
+		occupant.in_enclosed_vehicle = 1	//Useful for when you need to know if someone is in a mecho.
+>>>>>>> ce9ceb4... Add clickable action buttons from /TG/, rework a lot of backend and other improvements. (#7315)
```
Comment (lbnesquik) [inline code/game/mecha/mecha.dm:1502]: Rmv

**Comment by lbnesquik** [comment]
At the moment, the vorestation specific mechs have extra code  to be deleted. I could make a PR with it gone if that works?

**Comment by lbnesquik** [comment]
It might just work if  you merge this and #8368 right after.

**Comment by TheFurryFeline** [comment]
You could also just make a bench that merges into the upstream branch that's made for these mirrors, just to note.

**Pull request merged**